GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -267,6 +269,7 @@ async Task LoadLinksForRevisionAsync(GitRevision revision)
                         return;
                     }
 
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I think that all these `cancellationToken.ThrowIfCancellationRequested()`
should be directly in `StartAsyncDataLoad`. When we get to the state that all the execution is async-all-the-way-down then it will be clear that `cancellationToken.ThrowIfCancellationRequested()` are to be removed from `StartAsyncDataLoad`.
Putting `cancellationToken.ThrowIfCancellationRequested()` here brings the risk that they will stay here.

Reply:
I am afraid I don't quite understand what you're trying to say, sorry.

The way I understand it is as follows.
At the top of `StartAsyncDataLoad` we provision a new cancellation token. A previously issued token (if any) is cancelled, and aborts any pending operations. To achieve this we need `cancellationToken.ThrowIfCancellationRequested()` calls and pass the token to `this.SwitchToMainThreadAsync()` calls.

Reply:
What I am trying to say is that after your change `LoadAnnotatedTagInfoAsync` knows that it will be called in sequence.
Which I think it should not. Only the caller should know that, and it is its job to call `cancellationToken.ThrowIfCancellationRequested()`. If `LoadAnnotatedTagInfoAsync` was called in parallel then it would not be much sense to call `cancellationToken.ThrowIfCancellationRequested()` at the beginning, since we know that no cancellation could occur.
In addition there is a big chance that someone who makes it parallel again (when all the execution is async-all-the-way-down) will not remove this `cancellationToken.ThrowIfCancellationRequested()` at the beginning of `LoadAnnotatedTagInfoAsync`. The same applies to the other methods.

Reply:
I disagree, each method doesn't have a notion of the order it is called; the composition is still the function of `StartAsyncDataLoad`.

I am working on an update to the commit info control (though it is a glacial progress given my load) and planning to refactor/rework how data is populated and displayed to the user.

Reply:
I am afraid you still do not know what I am trying to say.

Reply:
I am not talking here about the order. I am talking about parallel vs sequential invocation. Putting cancellationToken.ThrowIfCancellationRequested() at the beginning of LoadAnnotatedTagInfoAsync makes only sense when `LoadAnnotatedTagInfoAsync ` knows it be invoked in a sequential way.

Reply:
Saying this in other words:
Lets assume I did the all-the-way-down-async and now I am changing StartAsyncDataLoad that it calls all the Load* functions in parallel. Does it make sense to leave `cancellationToken.ThrowIfCancellationRequested();` at the beginning of Load* functions? Or should them be removed?

Reply:
Updated